Tella/Data/Networking/Resources/ResourceService.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  ResourceService.swift
+//  Tella
+//
+//  Created by gus valbuena on 2/16/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class ResourceService {
+    private var cancellables: Set<AnyCancellable> = []
+    func getDownloadedResources(from appModel: MainAppModel) -> [DownloadedResourceCardViewModel] {
+        guard let resources = appModel.vaultManager.tellaData?.getResources() else {
+            return []
+        }
+
+        return resources.map { resource in
+            return createResourceCardViewModel(from: resource, in: appModel)
+        }
+    }
+
+    private func createResourceCardViewModel(
+        from resource: DownloadedResource, in appModel: MainAppModel
+    ) -> DownloadedResourceCardViewModel {
+        let selectedServer = appModel.vaultManager.tellaData?.tellaServers.value.first {
+            $0.id == resource.serverId
+        }
+
+        return DownloadedResourceCardViewModel(
+            id: resource.id,
+            externalId: resource.externalId,
+            title: resource.title,
+            fileName: resource.fileName,
+            serverName: selectedServer?.name ?? "",
+            size: resource.size,
+            createdAt: resource.createdAt
+        )
+    }
+
+    func getAvailableResources(appModel: MainAppModel, servers: [TellaServer]) -> AnyPublisher<
+        [ResourceCardViewModel], APIError
+    > {
+        // Logic to fetch available resources
+        let resourceRepository = ResourceRepository()
```

Comment:
I suggest to return a list of `Ressource domain model` instead of  `ResourceDTO` from `getResourcesByProject`  API and then map them  to `ResourceCardViewModel`

```
        let publishers = servers.map { server in
            resourceRepository.getResourcesByProject(server: server)
                .map { resource in
                    ResourceCardViewModel(resource: resource, server:server)
                }
        }

```